wandb/integration/keras/keras.py

```diff
@@ -958,5 +971,31 @@ def _save_model(self, epoch):
         # Was getting `RuntimeError: Unable to create link` in TF 1.13.1
         # also saw `TypeError: can't pickle _thread.RLock objects`
         except (ImportError, RuntimeError, TypeError) as e:
-            wandb.termerror("Can't save model, h5py returned error: %s" % e)
+            wandb.termerror(
+                "Can't save model in the h5py format. The model will be saved as "
+                "W&B Artifacts in the SavedModel format."
+            )
             self.save_model = False
+            self.save_model_as_artifact = True
+
+    def _save_model_as_artifact(self, epoch):
+        if wandb.run.disabled:
+            return
+        try:
+            # Save the model in the SavedModel format.
+            self.model.save(self.filepath[:-3], overwrite=True, save_format="tf")
+
+            # Log the model as artifact.
+            model_artifact = wandb.Artifact(f"model-{wandb.run.name}", type="model")
+            model_artifact.add_dir(self.filepath[:-3])
+            wandb.run.log_artifact(
+                model_artifact, aliases=["latest", "best", f"epoch_{epoch}"]
+            )
+
+            # Remove the SavedModel from wandb dir as we don't want to log it to save memory.
+            import shutil
+
+            shutil.rmtree(self.filepath[:-3])
+        except (ImportError, RuntimeError) as e:
+            wandb.termerror(f"Can't save model as artifact % {e}")
+            self.save_model_as_artifact = False
```

Comment:
Would you mind adding an explicit unit test to beef up codecov here?

Reply:
I don't think this except block is useful here. Removing it actually. 

If you defer can you let me know what possible exception it can hit? `ImportError` is something it can never hit imo. Not very sure about `RuntimeError`.